thunder/transforms/quantization.py

```diff
@@ -294,3 +295,214 @@ def transform_traces_pre_prologue(self, prologue_trace, computation_trace, epilo
+                        dtype=thunder.dtypes.to_dtype(qs["loraA.dtype"]),
+                        device=thunder.devices.to_device(device),
+                        requires_grad=requires_grad,
+                    )
+                    proxy_loraB = thunder.TensorProxy(
+                        name=f"{get_param.output.name}_loraB",
+                        shape=qs["loraB.shape"],
+                        dtype=thunder.dtypes.to_dtype(qs["loraB.dtype"]),
+                        device=thunder.devices.to_device(device),
+                        requires_grad=requires_grad,
+                    )
+                    new_bsyms.append(get_param.sym.bind(get_param.args[0], n_loraA, output=proxy_loraA))
+                    new_bsyms.append(get_param.sym.bind(get_param.args[0], n_loraB, output=proxy_loraB))
+
+                    add_trace_output(prologue_trace, proxy_loraA, subindex=0)
+                    add_trace_output(prologue_trace, proxy_loraB, subindex=0)
+
+                    new_compute_inputs.append(proxy_loraA)
+                    new_compute_inputs.append(proxy_loraB)
+                    # this is not good, because we will have several traces...
+                    additional_proxies[n_loraA] = proxy_loraA
+                    additional_proxies[n_loraB] = proxy_loraB
+
+        prologue_trace.bound_symbols[-1:-1] = new_bsyms
+
+        computation_proxy_map = {
+            csym.name: dict(
+                shape=psym.shape,
+                dtype=psym.dtype,
+            )
+            for psym, csym in zip(prologue_trace.bound_symbols[-1].args[0][0], computation_trace.args)
+            if psym.shape != csym.shape or psym.dtype != csym.dtype
+        }
+
+        new_computation_trace = trace_with_replaced_proxy_metadata(computation_trace, computation_proxy_map)
+        bound_symbols = new_computation_trace.bound_symbols
+        new_computation_trace.bound_symbols = []
+
+        new_computation_trace.args = (*new_computation_trace.args, *new_compute_inputs)
+        new_computation_trace._siginfo.args = [(a.name, None) for a in new_computation_trace.args]
+
+        with tracectx(new_computation_trace):
+            new_bindings = [
+                thunder.core.prims.unpack_trivial.bind(i, output=i, name=i.name) for i in new_compute_inputs
+            ]
+
+        for idx, bsym in enumerate(bound_symbols):
+            if bsym.sym != prims.unpack_trivial:
+                break
+            new_computation_trace.bound_symbols.append(bsym.from_bsym())
+        new_computation_trace.bound_symbols += new_bindings
+
+        for bsym in bound_symbols[idx:]:
+            if bsym.sym == thunder.torch.linear:
+                n = lora_linear_proxies[bsym.args[1].name]
+                with thunder.core.trace.tracectx(computation_trace):
+                    loraA_transpose_meta = prims.transpose.meta(additional_proxies[f"{n}.loraA"], (1, 0))
+                    loraA_bsym = prims.transpose.bind(
```

Comment:
I think the bsym should be automatically recorded when you call prims.transpose, no binding needed.

Reply:
I figured that the following does not work:
```
                    lora_a_bsym = prims.transpose(additional_proxies[f"{n}.lora_a"], (1, 0))
                    lora_b_bsym = prims.transpose(additional_proxies[f"{n}.lora_b"], (1, 0))
                    lora_meta_ = prims.matmul.meta(bsym.args[0], lora_a_bsym)
                    lora_bsym_ = prims.matmul.bind(bsym.args[0], lora_a_bsym, output=lora_meta_)
                    lora_meta = prims.matmul.meta(lora_bsym_.output, lora_b_bsym)
                    lora_bsym = prims.matmul.bind(lora_bsym_.output, lora_b_bsym, output=lora_meta)
                    original_weight_meta = prims.linear.meta(*bsym.args[:3])
                    original_weight = prims.linear.bind(*bsym.args[:3], output=original_weight_meta)

                    lora_sum_bsym = bsym.from_bsym(
                        sym=prims.add,
                        args=(original_weight.output, lora_bsym.output),
                    )

```
What I get from this is that, while I can remove the `.meta` by the simple `.transpose`, I don't think I can remove `.bind`. Or am I misinterpreting this somehow?